src/fileActions.jsx

```diff
@@ -603,3 +603,29 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        external: {
+            "content-disposition": `attachment; filename="${selected.name}"`,
+            "content-type": "application/octet-stream",
+        }
+    }));
+
+    const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
+    const url = prefix + "?" + query;
+    return new Promise((resolve, reject) => {
+        const iframe = document.createElement("iframe");
+        iframe.setAttribute("src", url);
+        iframe.setAttribute("hidden", "hidden");
+        iframe.addEventListener("load", event => {
+            document.body.removeChild(iframe);
+            resolve();
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-175-20240129-100727-b86fcbeb-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Reply:
I'm pretty sure I tested that the iframe was removed, I had a concern that we would keep creating iframes. But sure let's re-check.


![image](https://github.com/cockpit-project/cockpit-navigator/assets/67428/a082d002-9e59-4426-b832-72679767173f)

Works as expected:

![image](https://github.com/cockpit-project/cockpit-navigator/assets/67428/a6c38b27-e94e-40bc-bda3-3c562dd9b94a)

Reply:
Does the handler get called? I.e. did you add a console.log() to it?

Reply:
Yes, the console log if from the `load event`